### PR TITLE
[video_player_android] Add support for adding exoplayer extensions like ffmpeg to video_player on Android.

### PR DESCRIPTION
**Pull request by basemosama**
This pr adds support for adding [Exoplayer extensions](https://github.com/google/ExoPlayer/tree/release-v2/extensions) to video player on Android.

For example, We can add The [FFmpeg extension](https://exoplayer.dev/supported-formats.html#sample-formats) to support decoding a variety of different audio sample formats specially on old Android devices that don't have hardware decoders for audio formats like AC-3.

I have updated the example app with adding [ffmpeg extension](https://github.com/google/ExoPlayer/tree/release-v2/extensions/ffmpeg).

This Fixes https://github.com/flutter/flutter/issues/89038

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [] I added new tests to check the change I am making, or this PR is [test-exempt].
- [] All existing and new tests are passing.

### CODE REVIEW

File: packages/video_player/video_player_android/android/src/main/java/io/flutter/plugins/videoplayer/VideoPlayer.java [hunk 1]
```diff
@@ -76,8 +77,14 @@ final class VideoPlayer {
     this.textureEntry = textureEntry;
     this.options = options;
 
-    ExoPlayer exoPlayer = new ExoPlayer.Builder(context).build();
-    Uri uri = Uri.parse(dataSource);
+        final DefaultRenderersFactory renderersFactory = new DefaultRenderersFactory(context);
+        renderersFactory.setExtensionRendererMode(DefaultRenderersFactory.EXTENSION_RENDERER_MODE_PREFER);
+        renderersFactory.setEnableDecoderFallback(true);
```
[the remark this answers is not in the document]
Comment (basemosama) [reply packages/video_player/video_player_android/android/src/main/java/io/flutter/plugins/videoplayer/VideoPlayer.java:82]: I think having support for adding exoplayer extensions like ffmpeg audio decoder is important espacially for Android Tv box devices as most of them don't have hardware decoders for some formats like AC-3.

I think the best option here is to add a VideoPlayerAndroidOptions class inside VideoPlayerOptions and add the renderer mode and decoder fallback paramters into it.

**Comment by wtjperi2003** [comment]
> This pr adds support for adding [Exoplayer extensions](https://github.com/google/ExoPlayer/tree/release-v2/extensions) to video player on Android.
> 
> For example, We can add The [FFmpeg extension](https://exoplayer.dev/supported-formats.html#sample-formats) to support decoding a variety of different audio sample formats specially on old Android devices that don't have hardware decoders for audio formats like AC-3.
> 
> I have updated the example app with adding [ffmpeg extension](https://github.com/google/ExoPlayer/tree/release-v2/extensions/ffmpeg).
> 
> This Fixes [flutter/flutter#89038](https://github.com/flutter/flutter/issues/89038)
> 
> ## Pre-launch Checklist
> * [x]  I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
> * [x]  I read the [Tree Hygiene] wiki page, which explains my responsibilities.
> * [x]  I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
> * [x]  I signed the [CLA].
> * [x]  The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
> * [x]  I listed at least one issue that this PR fixes in the description above.
> * [x]  I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
> * [x]  I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
> * [x]  I updated/added relevant documentation (doc comments with `///`).
> * [] I added new tests to check the change I am making, or this PR is [test-exempt].
> * [] All existing and new tests are passing.

This is what I need. Can you solve prob

> This pr adds support for adding [Exoplayer extensions](https://github.com/google/ExoPlayer/tree/release-v2/extensions) to video player on Android.
> 
> For example, We can add The [FFmpeg extension](https://exoplayer.dev/supported-formats.html#sample-formats) to support decoding a variety of different audio sample formats specially on old Android devices that don't have hardware decoders for audio formats like AC-3.
> 
> I have updated the example app with adding [ffmpeg extension](https://github.com/google/ExoPlayer/tree/release-v2/extensions/ffmpeg).
> 
> This Fixes [flutter/flutter#89038](https://github.com/flutter/flutter/issues/89038)
> 
> ## Pre-launch Checklist
> * [x]  I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
> * [x]  I read the [Tree Hygiene] wiki page, which explains my responsibilities.
> * [x]  I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
> * [x]  I signed the [CLA].
> * [x]  The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
> * [x]  I listed at least one issue that this PR fixes in the description above.
> * [x]  I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
> * [x]  I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
> * [x]  I updated/added relevant documentation (doc comments with `///`).
> * [] I added new tests to check the change I am making, or this PR is [test-exempt].
> * [] All existing and new tests are passing.

This is what I need. Can you solve the problems?

**Comment by camsim99** [comment]
Note from triage: @basemosama are you planning on updating this PR to create VideoPlayerAndroidOptions or modify the platform interface?

**Comment by basemosama** [comment]
> Note from triage: @basemosama are you planning on updating this PR to create VideoPlayerAndroidOptions or modify the platform interface?

Yes, I will update it and add Options for android.

**Comment by reidbaker** [comment]
Thank you for your contribution. I'm going to close this PR for now since there are outstanding comments, just to get this off our PR review queue. Please don't hesitate to submit a new PR if you have the time to address the review comments. Thanks!

**Pull request closed without merging**